### DISCUSSION TITLE
Create a view that will contain a checkbox that makes it anchorable

### DISCUSSION THREAD

**Discussion by sboudreaux96** [eclipse-theia/theia#9556] [Q&A]
Hello - I would like to know how I would go about creating a view that will contain a checkbox. Once checked I would like the view to be locked in place (anchored) and when the checkbox isnt checked I would like to be able to move it freely.

**Comment by vince-fugnitto** [comment]
@sboudreaux96 thank you for the discussion, there is a similar example part of the repository:

- https://github.com/eclipse-theia/theia/tree/master/examples/api-samples/src/browser/view

Essentially, when the view behaves the following way:
- the view is uncloseable by default (checkbox unchecked)
- the view is closeable when the checkbox is checked

You can inspire yourself from it's implementation in order to keep the view fixed and controllable by the checkbox.

I'm not aware of anything that makes a `view` fixed however, so additional investigations are likely necessary. It may be worthwhile to support an option in the framework like closeable which makes views fixed :+1:
